### ISSUE TITLE
cursor problem

### ISSUE THREAD

**Issue by dreamend1993** [mankool0/GTFOHax#1]
after installed as README list 
dll file created successfully
but something wrong after inject
one when hide imgui window, two cursor appear in game
two world/enemy esp dont work
-----------------------------------------------
before all above
i was using dll download from ukc for days, it worked all ok
but now , it still show above problems
pls help me

**Comment by mankool0** [comment]
Did anything change on your end? Like game update or different injector? I just tried it and everything still works for me.
I'm not able to recreate your first issue with two cursors, so if you're able to describe the exact steps you take that'd be great.
For the esp the only way I can make it not work is injecting after starting a level.

**Comment by dreamend1993** [comment]
> Did anything change on your end? Like game update or different injector? I just tried it and everything still works for me. I'm not able to recreate your first issue with two cursors, so if you're able to describe the exact steps you take that'd be great. For the esp the only way I can make it not work is injecting after starting a level.

its solved , i inject my dll and your dll both by mistake , so that do not work well. very thank u for answering.
very very thank you . your code give me one window for learning how to write a hack.

for my reason to build this dll for my language in game some item translate to chinese so they do not draw in world esp.
i made a map for them. 

now im trying to do cpp2il on my own , for do something to my other games

thank you again

**Comment by dreamend1993** [comment]
could u tell me which commands to use to generate these il2cpp files thanks
may Il2CppInspector.exe -i GameAssembly.dll -m global-metadata.dat work ?

**Comment by dreamend1993** [comment]
its amazing for u can select right functions to hook to get data you want and call right function for certain purpose, may one day i can perform same like u. that's so wonderful

**Comment by mankool0** [comment]
> could u tell me which commands to use to generate these il2cpp files thanks may Il2CppInspector.exe -i GameAssembly.dll -m global-metadata.dat work ?

That should work, I haven't done it in awhile so I don't remember the commands off the top of my head.


> its amazing for u can select right functions to hook to get data you want and call right function for certain purpose, may one day i can perform same like u. that's so wonderful

You should look into using ghidra or ida pro. It's hard to guess what functions to hook just by looking at the names but if you decompile them it'll be a lot easier.

**Comment by dreamend1993** [comment]
> > could u tell me which commands to use to generate these il2cpp files thanks may Il2CppInspector.exe -i GameAssembly.dll -m global-metadata.dat work ?
> 
> That should work, I haven't done it in awhile so I don't remember the commands off the top of my head.
> 
> > its amazing for u can select right functions to hook to get data you want and call right function for certain purpose, may one day i can perform same like u. that's so wonderful
> 
> You should look into using ghidra or ida pro. It's hard to guess what functions to hook just by looking at the names but if you decompile them it'll be a lot easier.

yes but on my own use Il2CppInspector several times, but sometimes codes break , sometimes codes in functions are empty

**Comment by dreamend1993** [comment]
..\Il2CppInspector-cli.exe -i "F:\SteamLibrary\steamapps\common\GTFO\GameAssembly.dll" -m "F:\SteamLibrary\steamapps\common\GTFO\GTFO_Data\il2cpp_data\Metadata\global-metadata.dat"
Il2CppInspector Command-Line Edition
Version 2021.1
(c) 2017-2021 Katy Coe - www.djkaty.com - www.github.com/djkaty

Using plugin: IL2CPP API Discovery
Using plugin: Binary metadata field order deobfuscator
Using plugin: Metadata strings XOR decryptor
Using plugin: Binary file XOR decryptor
Detected metadata version 24.4
Container format: PE32+
Container endianness: Little
Architecture word size: 64-bit
Instruction set: x64
Global offset: 0x0000000180000C00
No symbol table present in binary file
No matches via code heuristics
Required structures acquired from data heuristics
CodeRegistration struct found at 0x0000000182DB8CC8 (file offset 0x02DB74C8)
MetadataRegistration struct found at 0x0000000182DBD3C0 (file offset 0x02DBBBC0)
The detected Il2CppCodeRegistration / Il2CppMetadataRegistration structs do not pass validation. This may mean that their fields have been re-ordered as a form of obfuscation and Il2CppInspector has not been able to restore the original order automatically. Consider re-ordering the fields in Il2CppBinaryClasses.cs and try again.
Analyze IL2CPP data: 1.61 sec


thats Il2CppInspector result , how il2cpp-functions make, i dont understand >_<

**Comment by dreamend1993** [comment]
Il2CppInspector error solved, used branch source
everything works well now
